src/main/java/com/jcraft/jsch/Channel.java

```diff
@@ -32,7 +32,7 @@
 import java.io.*;
 import java.util.Vector;
 
-public abstract class Channel implements Runnable{
+public abstract class Channel{
```

Comment:
I got your point about internal use. But as I read javadoc of runnable, is it not better to understand, that this instance is supposed to be executed by a thread and keep it as marker interface?

Reply:
My thought behind removing `Runnable` was to be able to change the `run()` method so that it was no longer public, since it isn't intended to be an API method that users would call. If this class implements `Runnable`, then the `run()` method has to be public, as it isn't possible to reduce its visibility.

Also, the primary purpose of `Channel` objects isn't to be a `Runnable` itself, at least from a user's perspective: that's just an internal implementation detail within JSch, with how it handles individual SSH channels within each session.

Reply:
yes I understand your point. Now its a negotiation whether it is better to hide internals or to show to the developer, that the code of `run()` method is supposed to be run by a `Thread`. I think both ways have their arguments and there is no right or wrong. Let's keep you change for now. Maybe someone else will stumble on this in the future and then we can discuss further.